### PR TITLE
Update the local DBs for REJECTED resources

### LINKED ISSUE #20388: Properly update the vulnerability detector internal DBs in case a previously PUBLISHED document becomes REJECTED
## Description

All the new elements that the `contentManager` publishes to the `databaseFeedManager` are processeced in `storeModel` to update the `descriptions`, `remediations`, `candidates` and `translations` internal DBs.

The case when a CVE 5.0 document is first informed as `PUBLISHED` but then in an update changes to `REJECTED` has to be properly handled to avoid leaving invalid information in the mentioned DBs.

## DoD

- [x] Share a comment with the analysis of the situation and the possible solution
- [x] Update the required orchestration to handle the described situation
- [x] Perform manual tests, changing a CVE from one state to another. Include other tests if required
- [x] Add UT
- [x] CI Pass

### PR DESCRIPTION
**Pull request by pereyra-m**
|Related issue|
|---|
|Closes #20388|

## Description

This PR solves a synchronization problem with the local DBs when the published documents become `REJECTED` after being `PUBLISHED`.

The descriptions and remediations have a simple design and the entries are removed directly. For candidates, it was required to write extra information in the DB (in another column family) to keep track of the entries related to a CVE and remove them later.

The `rocksdDBWrapper` was updated with all the required methods for batch and column family operations.

## Tests

To verify the changes, a single document was processed with the `databaseFeedManager` test tool

<details><summary>Details</summary>
<p>

```json
{
    "data": [
        {
            "offset": 7001,
            "type": "create",
            "version": 1,
            "context": "all_vendors_071123",
            "resource": "CVE-2003-0380",
            "payload": {
                "containers": {
                    "adp": [
                        {
                            "affected": [
                                {
                                    "platforms": [
                                        "bookworm",
                                        "bullseye",
                                        "buster",
                                        "sid",
                                        "trixie"
                                    ],
                                    "product": "atftp",
                                    "vendor": "debian",
                                    "versions": [
                                        {
                                            "lessThan": "0.6.2",
                                            "status": "affected",
                                            "version": "0",
                                            "versionType": "deb"
                                        }
                                    ]
                                },
                                {
                                    "platforms": [
                                        "bookworm",
                                        "bullseye",
                                        "buster",
                                        "sid",
                                        "trixie"
                                    ],
                                    "product": "atftpd",
                                    "vendor": "debian",
                                    "versions": [
                                        {
                                            "lessThan": "0.6.2",
                                            "status": "affected",
                                            "version": "0",
                                            "versionType": "deb"
                                        }
                                    ]
                                }
                            ],
                            "metrics": [
                                {
                                    "other": {
                                        "content": {
                                            "data": "{\"description\":\"not yet assigned\"}"
                                        },
                                        "type": "Unknown"
                                    }
                                }
                            ],
                            "providerMetadata": {
                                "orgId": "79363d38-fa19-49d1-9214-5f28da3f3ac5",
                                "shortName": "debian"
                            },
                            "references": [
                                {
                                    "url": "https://security-tracker.debian.org/tracker/CVE-2003-0380"
                                }
                            ]
                        }
                    ],
                    "cna": {
                        "affected": [
                            {
                                "cpes": [
                                    "cpe:2.3:a:atftpd:atftpd:0.6.0:*:*:*:*:*:*:*",
                                    "cpe:2.3:a:atftpd:atftpd:0.6.1.1:*:*:*:*:*:*:*"
                                ],
                                "defaultStatus": "unaffected",
                                "product": "atftpd",
                                "vendor": "atftpd",
                                "versions": [
                                    {
                                        "status": "affected",
                                        "version": "0.6.0"
                                    },
                                    {
                                        "status": "affected",
                                        "version": "0.6.1.1"
                                    }
                                ]
                            }
                        ],
                        "descriptions": [
                            {
                                "lang": "en",
                                "value": "Buffer overflow in atftp daemon (atftpd) 0.6.1 and earlier, and possibly later versions, allows remote attackers to cause a denial of service (crash) and possibly execute arbitrary code via a long filename."
                            },
                            {
                                "lang": "es",
                                "value": "Desbordamiento de búfer en el demonio atftp (atftpd) 0.6.1 y anteriores, y posiblemente versiones posteriores, permite a atacantes remotos ejecutar código arbitrario mediante un nombre de fichero largo."
                            }
                        ],
                        "metrics": [
                            {
                                "cvssV2_0": {
                                    "accessComplexity": "LOW",
                                    "accessVector": "NETWORK",
                                    "authentication": "NONE",
                                    "availabilityImpact": "PARTIAL",
                                    "baseScore": 7.5,
                                    "confidentialityImpact": "PARTIAL",
                                    "integrityImpact": "PARTIAL",
                                    "vectorString": "AV:N/AC:L/Au:N/C:P/I:P/A:P",
                                    "version": "2.0"
                                },
                                "format": "CVSS"
                            }
                        ],
                        "problemTypes": [
                            {
                                "descriptions": [
                                    {
                                        "description": "NVD-CWE-Other",
                                        "lang": "en"
                                    }
                                ]
                            }
                        ],
                        "providerMetadata": {
                            "dateUpdated": "2008-09-05T20:34:11Z",
                            "orgId": "00000000-0000-4000-A000-000000000003",
                            "shortName": "nvd"
                        },
                        "references": [
                            {
                                "tags": [
                                    "exploit",
                                    "patch",
                                    "vendor-advisory"
                                ],
                                "url": "http://archives.neohapsis.com/archives/bugtraq/2003-06/0056.html"
                            },
                            {
                                "tags": [
                                    "patch",
                                    "vendor-advisory"
                                ],
                                "url": "http://www.debian.org/security/2003/dsa-314"
                            },
                            {
                                "tags": [
                                    "vendor-advisory"
                                ],
                                "url": "http://www.securityfocus.com/archive/82/323886/2003-06-02/2003-06-08/0"
                            }
                        ]
                    }
                },
                "cveMetadata": {
                    "assignerOrgId": "00000000-0000-4000-A000-000000000003",
                    "assignerShortName": "nvd",
                    "cveId": "CVE-2003-0380",
                    "datePublished": "2003-07-02T04:00:00Z",
                    "dateUpdated": "2008-09-05T20:34:11Z",
                    "state": "PUBLISHED"
                },
                "dataType": "CVE_RECORD",
                "dataVersion": "5.0"
            }
        }
    ]
}

```

</p>
</details> 

The DBs contain the expected entries

<details><summary>Details</summary>
<p>

```
root@ubuntu-jammy:/home/vagrant/wazuh/src/build/wazuh_modules/vulnerability_scanner/testtool/databaseFeedManager# ./database_feed_manager_testtool -c config.json  -i /home/vagrant/contents/
Press enter to stop the tool...
wazuh-modulesd:vulnerability-scanner->/home/vagrant/wazuh/src/wazuh_modules/vulnerability_scanner/src/databaseFeedManager/databaseFeedManager.hpp:180 Processing message
wazuh-modulesd:vulnerability-scanner->/home/vagrant/wazuh/src/wazuh_modules/vulnerability_scanner/src/databaseFeedManager/databaseFeedManager.hpp:113 Processing file: /home/vagrant/contents/test.json
wazuh-modulesd:vulnerability-scanner->/home/vagrant/wazuh/src/wazuh_modules/vulnerability_scanner/src/databaseFeedManager/databaseFeedManager.hpp:182 Message processed

root@ubuntu-jammy:/home/vagrant/wazuh/src/build/wazuh_modules/vulnerability_scanner/testtool/databaseFeedManager# ./rocks_db_query_
testtool -d queue/vd/candidates/debian -f /home/vagrant/wazuh/src/wazuh_modules/vulnerability_scanner/schemas/vulnerabilityCandidat
e.fbs -c cveId
/home/vagrant/wazuh/src/external/rocksdb/include/rocksdb/slice.h:114:42: runtime error: null pointer passed as argument 2, which is declared to never be null
atftp_CVE-2003-0380 ==> {
  "candidates": [
    {
      "cveId": "CVE-2003-0380",
      "platforms": [
        "bookworm",
        "bullseye",
        "buster",
        "sid",
        "trixie"
      ],
      "versions": [
        {
          "version": "0",
          "lessThan": "0.6.2",
          "versionType": "deb"
        }
      ]
    }
  ]
}

atftpd_CVE-2003-0380 ==> {
  "candidates": [
    {
      "cveId": "CVE-2003-0380",
      "platforms": [
        "bookworm",
        "bullseye",
        "buster",
        "sid",
        "trixie"
      ],
      "versions": [
        {
          "version": "0",
          "lessThan": "0.6.2",
          "versionType": "deb"
        }
      ]
    }
  ]
}

root@ubuntu-jammy:/home/vagrant/wazuh/src/build/wazuh_modules/vulnerability_scanner/testtool/databaseFeedManager# /home/vagrant/ldb
 --db=./queue/vd/candidates/debian scan --try_load_options --column_family=cveId
CVE-2003-0380_atftp : atftp_CVE-2003-0380
CVE-2003-0380_atftpd : atftpd_CVE-2003-0380

root@ubuntu-jammy:/home/vagrant/wazuh/src/build/wazuh_modules/vulnerability_scanner/testtool/databaseFeedManager# ./rocks_db_query_testtool -d queue/vd/candidates/nvd -f /home/vagrant/wazuh/src/wazuh_modules/vulnerability_scanner/schemas/vulnerabilityCandidate.f
bs -c cveId
/home/vagrant/wazuh/src/external/rocksdb/include/rocksdb/slice.h:114:42: runtime error: null pointer passed as argument 2, which is declared to never be null
atftpd_CVE-2003-0380 ==> {
  "candidates": [
    {
      "cveId": "CVE-2003-0380",
      "defaultStatus": "unaffected",
      "platforms": [

      ],
      "versions": [
        {
          "version": "0.6.0"
        },
        {
          "version": "0.6.1.1"
        }
      ]
    }
  ]
}

root@ubuntu-jammy:/home/vagrant/wazuh/src/build/wazuh_modules/vulnerability_scanner/testtool/databaseFeedManager# /home/vagrant/ldb --db=./queue/vd/candidates/nvd scan --try_load_options --column_family=cveId
CVE-2003-0380_atftpd : atftpd_CVE-2003-0380

```

</p>
</details> 

Then, an update message was processed and the document became `REJECTED`

<details><summary>Details</summary>
<p>

```json
{
    "data": [
        {
            "offset": 7001,
            "type": "update",
            "version": 1,
            "context": "all_vendors_071123",
            "resource": "CVE-2003-0380",
            "operations": [
                {
                    "op": "replace",
                    "path": "/cveMetadata/state",
                    "value": "REJECTED"
                }
            ]
        }
    ]
}

```

</p>
</details> 

We can verify that the entries were removed

<details><summary>Details</summary>
<p>

```
root@ubuntu-jammy:/home/vagrant/wazuh/src/build/wazuh_modules/vulnerability_scanner/testtool/databaseFeedManager# ./database_feed_m
anager_testtool -c config.json  -i /home/vagrant/contents/
Press enter to stop the tool...
wazuh-modulesd:vulnerability-scanner->/home/vagrant/wazuh/src/wazuh_modules/vulnerability_scanner/src/databaseFeedManager/databaseFeedManager.hpp:180 Processing message
wazuh-modulesd:vulnerability-scanner->/home/vagrant/wazuh/src/wazuh_modules/vulnerability_scanner/src/databaseFeedManager/databaseFeedManager.hpp:113 Processing file: /home/vagrant/contents/test.json
wazuh-modulesd:vulnerability-scanner->/home/vagrant/wazuh/src/wazuh_modules/vulnerability_scanner/src/databaseFeedManager/databaseFeedManager.hpp:182 Message processed

root@ubuntu-jammy:/home/vagrant/wazuh/src/build/wazuh_modules/vulnerability_scanner/testtool/databaseFeedManager# ./rocks_db_query_testtool -d queue/vd/candidates/debian/ -f /home/vagrant/wazuh/src/wazuh_modules/vulnerability_scanner/schemas/vulnerabilityCandida
te.fbs -c cveId
root@ubuntu-jammy:/home/vagrant/wazuh/src/build/wazuh_modules/vulnerability_scanner/testtool/databaseFeedManager# ./rocks_db_query_testtool -d queue/vd/candidates/nvd -f /home/vagrant/wazuh/src/wazuh_modules/vulnerability_scanner/schemas/vulnerabilityCandidate.f
bs -c cveId
root@ubuntu-jammy:/home/vagrant/wazuh/src/build/wazuh_modules/vulnerability_scanner/testtool/databaseFeedManager# /home/vagrant/ldb --db=./queue/vd/candidates/nvd scan --try_load_options --column_family=cveId
root@ubuntu-jammy:/home/vagrant/wazuh/src/build/wazuh_modules/vulnerability_scanner/testtool/databaseFeedManager# /home/vagrant/ldb --db=./queue/vd/candidates/debian scan --try_load_options --column_family=cveId
```

</p>
</details> 

Another test carried out consisted in removing only one entry

<details><summary>Details</summary>
<p>

```json
{
    "data": [
        {
            "offset": 7001,
            "type": "update",
            "version": 1,
            "context": "all_vendors_071123",
            "resource": "CVE-2003-0380",
            "operations": [
                {
                    "op": "remove",
                    "path": "/containers/adp/0/affected/0",
                    "value": "REJECTED"
                }
            ]
        }
    ]
}

```

</p>
</details> 

And we can verify that only the removed entry was deleted

<details><summary>Details</summary>
<p>

```
root@ubuntu-jammy:/home/vagrant/wazuh/src/build/wazuh_modules/vulnerability_scanner/testtool/databaseFeedManager# ./database_feed_manager_testtool -c config.json  -i /home/vagrant/contents/
Press enter to stop the tool...
wazuh-modulesd:vulnerability-scanner->/home/vagrant/wazuh/src/wazuh_modules/vulnerability_scanner/src/databaseFeedManager/databaseFeedManager.hpp:180 Processing message
wazuh-modulesd:vulnerability-scanner->/home/vagrant/wazuh/src/wazuh_modules/vulnerability_scanner/src/databaseFeedManager/databaseFeedManager.hpp:113 Processing file: /home/vagrant/contents/test.json
wazuh-modulesd:vulnerability-scanner->/home/vagrant/wazuh/src/wazuh_modules/vulnerability_scanner/src/databaseFeedManager/databaseFeedManager.hpp:182 Message processed

root@ubuntu-jammy:/home/vagrant/wazuh/src/build/wazuh_modules/vulnerability_scanner/testtool/databaseFeedManager# ./rocks_db_query_testtool -d queue/vd/candidates/debian/ -f /home/vagrant/wazuh/src/wazuh_modules/vulnerability_scanner/schemas/vulnerabilityCandida
te.fbs -c cveId
/home/vagrant/wazuh/src/external/rocksdb/include/rocksdb/slice.h:114:42: runtime error: null pointer passed as argument 2, which is declared to never be null
atftpd_CVE-2003-0380 ==> {
  "candidates": [
    {
      "cveId": "CVE-2003-0380",
      "platforms": [
        "bookworm",
        "bullseye",
        "buster",
        "sid",
        "trixie"
      ],
      "versions": [
        {
          "version": "0",
          "lessThan": "0.6.2",
          "versionType": "deb"
        }
      ]
    }
  ]
}

root@ubuntu-jammy:/home/vagrant/wazuh/src/build/wazuh_modules/vulnerability_scanner/testtool/databaseFeedManager# /home/vagrant/ldb --db=./queue/vd/candidates/debian scan --try_load_options --column_family=cveId
CVE-2003-0380_atftpd : atftpd_CVE-2003-0380

root@ubuntu-jammy:/home/vagrant/wazuh/src/build/wazuh_modules/vulnerability_scanner/testtool/databaseFeedManager# ./rocks_db_query_testtool -d queue/vd/candidates/nvd -f /home/vagrant/wazuh/src/wazuh_modules/vulnerability_scanner/schemas/vulnerabilityCandidate.f
bs -c cveId
/home/vagrant/wazuh/src/external/rocksdb/include/rocksdb/slice.h:114:42: runtime error: null pointer passed as argument 2, which is declared to never be null
atftpd_CVE-2003-0380 ==> {
  "candidates": [
    {
      "cveId": "CVE-2003-0380",
      "defaultStatus": "unaffected",
      "platforms": [

      ],
      "versions": [
        {
          "version": "0.6.0"
        },
        {
          "version": "0.6.1.1"
        }
      ]
    }
  ]
}

root@ubuntu-jammy:/home/vagrant/wazuh/src/build/wazuh_modules/vulnerability_scanner/testtool/databaseFeedManager# /home/vagrant/ldb --db=./queue/vd/candidates/nvd scan --try_load_options --column_family=cveId
CVE-2003-0380_atftpd : atftpd_CVE-2003-0380
```

</p>
</details> 

Minimum checks required
- Compilation without warnings in every supported platform
  - [x] Linux

- [x] Added unit tests (for new features)

### CODE REVIEW

**Review by MiguelazoDS — COMMENTED** [review]
GJ! 

Filling the database with JSON information.
<details><summary>Show image</summary>

image

</details>

Checking rocksDB 

<details><summary>Show image</summary>

image

</details>

Rejected CVE 

<details><summary>Show image</summary>

image

</details>

The information about CVE-2003-0380 is no longer present in the database
<details><summary>Show image</summary>

image

</details>

File: src/wazuh_modules/vulnerability_scanner/testtool/rocksDBQuery/main.cpp [hunk 1]
```diff
@@ -21,7 +21,7 @@ int main(const int argc, const char** argv)
     {
         CmdLineArgs cmdLineArgs(argc, argv);
 
-        auto rocksDB = Utils::RocksDBWrapper(cmdLineArgs.getDBPath());
```
Comment (MiguelazoDS) [inline src/wazuh_modules/vulnerability_scanner/testtool/rocksDBQuery/main.cpp:24]: I've got this error. I don't think is very important in this tool, but do you know the origin?
image

File: src/wazuh_modules/vulnerability_scanner/testtool/rocksDBQuery/main.cpp [hunk 2]
```diff
@@ -21,7 +21,7 @@ int main(const int argc, const char** argv)
...
+        auto rocksDB = Utils::RocksDBWrapper(cmdLineArgs.getDBPath(), cmdLineArgs.getColumnFamilies());
```
Comment (MiguelazoDS) [inline src/wazuh_modules/vulnerability_scanner/testtool/rocksDBQuery/main.cpp:24]: I'm getting two duplicate outputs, is that correct? again, I don't think this is a stopper for this implementation.

**Review by MiguelazoDS — DISMISSED** [review]

File: src/wazuh_modules/vulnerability_scanner/testtool/rocksDBQuery/main.cpp (hunk 2 above)
Comment (MiguelazoDS) [reply src/wazuh_modules/vulnerability_scanner/testtool/rocksDBQuery/main.cpp:24]: Too late to see the difference! Thanks!

File: src/wazuh_modules/vulnerability_scanner/testtool/rocksDBQuery/main.cpp (hunk 1 above)
Comment (pereyra-m) [reply src/wazuh_modules/vulnerability_scanner/testtool/rocksDBQuery/main.cpp:24]: I'll look at this again, thank you.

Comment (pereyra-m) [reply src/wazuh_modules/vulnerability_scanner/testtool/rocksDBQuery/main.cpp:24]: Hi again @MiguelazoDS , I've solved this warning in the last commit.

**Comment by pereyra-m** [comment]
The development was moved to #21410.

**Pull request closed without merging**